### PR TITLE
(352) show landing page only when campaign is inprogress

### PR DESCRIPTION
**Pull request by theflexbird**
#352 
It is dependent from #392.

### CODE REVIEW

File: src/server/routes/templates.js [hunk 1]
```diff
@@ -72,7 +74,7 @@ TemplateWebApi.prototype.renderPublicTemplate = function(type, req, res)
     const transition = req.query.transition;
     const baseUrl = '//' + req.headers['host'];
     const invitationCode = req.query.invitationCode;
-    const language = req.query.lang || req.cookies.lang;
+    const language = getLanguage(req);
```
Comment (kwierchris) [inline src/server/routes/templates.js]: The language has already been determined correctly.

File: src/server/routes/templates.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
 const Promise = require('bluebird');
 const extend = require('extend');
 const Templates = require('../api/Templates');
+const i18n = require('./i18n');
+const getLanguage = require('../utils/getLanguage');
```
Comment (kwierchris) [inline src/server/routes/templates.js]: Doing this adds another layer of not required complexity.

File: src/server/utils/getLanguage.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+module.exports = (req, altLanguage) => {
```
Comment (kwierchris) [inline src/server/utils/getLanguage.js]: A closure this is not a good idea as it requires certain variables to be available in the outside context and everyone using it would exactly have to know what it needs.

File: src/server/api/Templates.js [hunk 4]
```diff
@@ -4,88 +4,132 @@ const Handlebars = require('handlebars');
...
+    const customerPromise = this.serviceClient.get('customer', `/api/customers/${customerId}`, true).spread(c => c);
+    const contactPromise = contactId ? CampaignContact.findById(contactId) : Promise.resolve({});
+
+    return Promise.all([campaignPromise, customerPromise, contactPromise]).spread((campaign, customer, contact) => {
+      if (!campaign) {
+        console.log('The requested campaign does not exist.');
+        throw {
+          name: 'workflow.error.campaignTitle',
+          message: 'workflow.error.campaignMsg'
+        };
```
Comment (nczeranka) [inline src/server/api/Templates.js]: why did you remove the Error?

File: src/server/api/Templates.js [hunk 5]
```diff
@@ -4,88 +4,132 @@ const Handlebars = require('handlebars');
...
+      }
+      if (!customer) {
+        console.log('The requested customer does not exist.');
+        throw {
+          name: 'workflow.error.customerTitle',
+          message: 'workflow.error.customerMsg'
+        };
```
Comment (nczeranka) [inline src/server/api/Templates.js]: see above.

File: src/server/api/Templates.js [hunk 6]
```diff
@@ -4,88 +4,132 @@ const Handlebars = require('handlebars');
...
+      return Template.findOne({where: templateWhere})
+        .then(template => [template, campaign, customer, contact]);
+    })
+      .spread((template, campaign, customer, contact) => {
+        if (!template) {
+          console.log('The template required by the requested campaign could not be found.');
+          throw {
+            name: 'workflow.error.internalTitle',
+            message: 'workflow.error.internalMsg'
+          };
```
Comment (nczeranka) [inline src/server/api/Templates.js]: see above.

File: src/server/api/Templates.js [hunk 7]
```diff
@@ -4,88 +4,132 @@ const Handlebars = require('handlebars');
...
+        const templateValues = {
+          campaign,
+          customer,
+          contact,
+          transition,
+          invitationCode,
+          externalUrl,
+          blobUrl,
+          language
+        };
```
Comment (nczeranka) [inline src/server/api/Templates.js]: Why did you remove emailOpenTrack?
This render method is used from workflow.js to generate the emails, too. There we need this feature.

File: src/server/api/Templates.js [hunk 8]
```diff
@@ -4,88 +4,132 @@ const Handlebars = require('handlebars');
...
+          const applyTransition = allowed && allowed.indexOf(transition) ? transition : null;
+
+          if (applyTransition) {
+            return contact.updateAttributes({
+              status: transition,
+              lastStatusChange: new Date()
+            })
+              .then(() => ({result, templateValues}));
+          }
+        }
```
Comment (nczeranka) [inline src/server/api/Templates.js]: What is the sense of setting the status in this render method?
Rendering itself is not allowed to change the status - e.g. rendering the email template as part of the workflow.js should not have any effect on the status!
Status is set when the endpoint is triggered (see server/routes/templates.js) and not here.

**Comment by nczeranka** [comment]
Fixed via #488
